runtime/autoload/ccomplete.lua

```diff
@@ -0,0 +1,892 @@
+----------------------------------------
+-- This file is generated via github.com/tjdevries/vim9jit
+-- For any bugs, please first consider reporting there.
+----------------------------------------
+
+local NVIM9 = require('_vim9script')
+local __VIM9_MODULE = {}
+local prepended = nil
+local grepCache = nil
+local Complete = nil
+local GetAddition = nil
+local Tag2item = nil
+local Dict2info = nil
+local ParseTagline = nil
+local Tagline2item = nil
+local Tagcmd2extra = nil
+local Nextitem = nil
+local StructMembers = nil
+local SearchMembers = nil
+-- vim9script
+
+-- # Vim completion script
+-- # Language:     C
+-- # Maintainer:   Bram Moolenaar <Bram@vim.org>
+-- #		Rewritten in Vim9 script by github user lacygoill
+-- # Last Change:  2022 Jan 31
+
+prepended = ''
+grepCache = vim.empty_dict()
+
+-- # This function is used for the 'omnifunc' option.
+
+Complete = function(findstart, abase)
+  findstart = NVIM9.bool(findstart)
+  if NVIM9.bool(findstart) then
+    -- # Locate the start of the item, including ".", "->" and "[...]".
+    local line = NVIM9.fn['getline']('.')
+    local start = NVIM9.fn['charcol']('.') - 1
+    local lastword = -1
+    while start > 0 do
+      if NVIM9.ops['RegexpMatches'](NVIM9.index(line, NVIM9.ops['Minus'](start, 1)), '\\w') then
+        start = start - 1
+      elseif
+        NVIM9.bool(
+          NVIM9.ops['RegexpMatches'](NVIM9.index(line, NVIM9.ops['Minus'](start, 1)), '\\.')
+        )
+      then
+        if lastword == -1 then
+          lastword = start
+        end
+        start = start - 1
+      elseif
+        NVIM9.bool(
+          start > 1
+            and NVIM9.index(line, NVIM9.ops['Minus'](start, 2)) == '-'
+            and NVIM9.index(line, NVIM9.ops['Minus'](start, 1)) == '>'
+        )
+      then
+        if lastword == -1 then
+          lastword = start
+        end
+        start = NVIM9.ops['Minus'](start, 2)
+      elseif NVIM9.bool(NVIM9.index(line, NVIM9.ops['Minus'](start, 1)) == ']') then
+        -- # Skip over [...].
+        local n = 0
+        start = start - 1
+        while start > 0 do
+          start = start - 1
+          if NVIM9.index(line, start) == '[' then
+            if n == 0 then
+              break
+            end
+            n = n - 1
+          elseif NVIM9.bool(NVIM9.index(line, start) == ']') then
+            n = n + 1
+          end
+        end
+      else
+        break
+      end
+    end
+
+    -- # Return the column of the last word, which is going to be changed.
+    -- # Remember the text that comes before it in prepended.
+    if lastword == -1 then
+      prepended = ''
+      return NVIM9.fn['byteidx'](line, start)
+    end
+    prepended = NVIM9.slice(line, start, NVIM9.ops['Minus'](lastword, 1))
+    return NVIM9.fn['byteidx'](line, lastword)
+  end
+
+  -- # Return list of matches.
+
```

Comment:
can these blank lines after comments be avoided?

Reply:
i'll look into it

Reply:
made an issue upstream that could work on this later